checker/src/environment.rs

```diff
@@ -58,10 +58,14 @@ impl Environment {
         if let Some((join_condition, true_path, false_path)) = self.try_to_split(&path) {
             // If path is an abstraction that can match more than one path, we need to do weak updates.
             let top = abstract_value::TOP;
-            let true_val = join_condition
-                .conditional_expression(&value, self.value_at(&true_path).unwrap_or(&top));
-            let false_val = join_condition
-                .conditional_expression(self.value_at(&false_path).unwrap_or(&top), &value);
+            let true_val = join_condition.clone().conditional_expression(
```

Comment:
You have to clone the join_condition to check it?

Reply:
I'm afraid so. join_condition is used again later in context where it is moved. I.e. I need two copies of it.